examples/container-service-extension/3.7-cse-step1.tf

```diff
@@ -0,0 +1,269 @@
+    prefix_length = "24"
+    primary_ip    = "10.0.0.2"
+
+    allocated_ips {
+      start_address = "10.0.0.2"
+      end_address   = "10.0.0.254"
+    }
+  }
+
+  depends_on = [vcd_org_vdc.cse_vdc]
+}
+
+# Routed network for the Kubernetes cluster.
+
+resource "vcd_network_routed_v2" "cse_routed" {
+  org         = vcd_org.cse_org.name
+  name        = "cse_routed_net"
+  description = "My routed Org VDC network backed by NSX-T"
+
+  edge_gateway_id = vcd_nsxt_edgegateway.cse_egw.id
+
+  gateway       = "192.168.7.1"
+  prefix_length = 24
+
+  static_ip_pool {
+    start_address = "192.168.7.2"
+    end_address   = "192.168.7.100"
+  }
+
+  dns1 = "8.8.8.8"
+  dns2 = "8.8.8.4"
+}
+
+# NAT rule to map traffic to internal network IPs
+
+resource "vcd_nsxt_nat_rule" "snat" {
+  org             = vcd_org.cse_org.name
+  edge_gateway_id = vcd_nsxt_edgegateway.cse_egw.id
+
+  name        = "SNAT rule"
+  rule_type   = "SNAT"
+  description = "description"
+
+  external_address = "10.0.0.3"
+  internal_address = "192.168.7.0/24"
+  logging          = true
+}
+
+# Cluster requires network traffic is open, to download required dependencies to create nodes. Adapt this firewall
+# rule to your organization security requirements, as this is just an example.
+
+resource "vcd_nsxt_firewall" "firewall" {
+  org             = vcd_org.cse_org.name
+  edge_gateway_id = vcd_nsxt_edgegateway.cse_egw.id
+
+  rule {
+    action      = "ALLOW"
+    name        = "allow all IPv4 traffic"
+    direction   = "IN_OUT"
+    ip_protocol = "IPV4"
+  }
+}
+
+# Catalog to upload the TKGm OVAs
+
+data "vcd_storage_profile" "cse_sp" {
+  org  = vcd_org.cse_org.name
+  vdc  = vcd_org_vdc.cse_vdc.name
+  name = "*"
+
+  depends_on = [vcd_org.cse_org, vcd_org_vdc.cse_vdc]
+}
+
+resource "vcd_catalog" "cat-cse" {
+  org         = vcd_org.cse_org.name
+  name        = "cat-cse"
+  description = "CSE catalog"
+
+  storage_profile_id = data.vcd_storage_profile.cse_sp.id
+
+  delete_force     = "true"
+  delete_recursive = "true"
+  depends_on       = [vcd_org_vdc.cse_vdc]
+}
+
+# TKGm OVA upload. The `catalog_item_metadata` is required for CSE to detect the OVAs.
+
+resource "vcd_catalog_item" "tkgm_ova" {
+  org     = vcd_org.cse_org.name
+  catalog = vcd_catalog.cat-cse.name
+
+  name                 = "ubuntu-2004-kube-v1.21.2+vmware.1-tkg.1-7832907791984498322"
+  description          = "ubuntu-2004-kube-v1.21.2+vmware.1-tkg.1-7832907791984498322"
+  ova_path             = "/Users/bob/CSE/TKGm/ubuntu-2004-kube-v1.21.2+vmware.1-tkg.1-7832907791984498322.iso"
+  upload_piece_size    = 100
+  show_upload_progress = true
+
+  catalog_item_metadata = {
+    "kind"               = "TKGm"
+    "kubernetes"         = "TKGm"
+    "kubernetes_version" = "v1.21.2+vmware.1"
+    "name"               = "ubuntu-2004-kube-v1.21.2+vmware.1-tkg.1-7832907791984498322"
+    "os"                 = "ubuntu"
+    "revision"           = "1"
+  }
+}
+
+# AVI configuration for Kubernetes services
```

Comment:
Would be good to explain the "why" it's needed too.

Reply:
Done, thanks!